renku_notebooks/api/notebooks.py

```diff
@@ -203,29 +287,37 @@ def launch_notebook(
       tags:
         - servers
     """
-    server_name = make_server_name(user.safe_username, namespace, project, branch, commit_sha)
+    if renku_2:
```

Comment:
maybe cleaner (and easier to remove later) if renku and renku_2 logic was handled in two separate functions

Reply:
The common logic is too much and it would be difficult to use a template function pattern here. I've moved the uncommon logic to the caller functions.